presto-native-execution/presto_cpp/main/tests/TaskManagerTest.cpp

```diff
@@ -823,4 +830,59 @@ TEST_F(TaskManagerTest, aggregationSpill) {
   }
 }
 
+TEST_F(TaskManagerTest, createOrUpdateTask) {
+    auto filePaths = makeFilePaths(5);
+    auto vectors = makeVectors(filePaths.size(), 1'000);
+    for (int i = 0; i < filePaths.size(); i++) {
+        writeToFile(filePaths[i]->path, vectors[i]);
+    }
+    duckDbQueryRunner_.createTable("tmp", vectors);
+    auto planFragment = exec::test::PlanBuilder()
+            .tableScan(rowType_)
+            .filter("c0 % 5 = 0")
+            .partitionedOutput({}, 1, {"c0", "c1"})
+            .planFragment();
+
+    long splitSequenceId{0};
+    auto scanTaskId = "scan.0.0.1";
+    auto maxSize = protocol::DataSize("32MB");
+    auto maxWait = protocol::Duration("2s");
+    // Simulate taskmanager accessing a buffer for taskId entry
+    // erased in PartitionedOutputBufferManager Map. Task is only cancelled and
+    // the buffer not found exception trickles up to the user.
+    auto resultRequestState = http::CallbackRequestHandlerState::create();
+    try {
+        // Task hasn't started, getResults will add a pending resultRequest to prestoTask.
+        taskManager_->getResults(scanTaskId, 0, splitSequenceId,
+                                 maxSize, maxWait, resultRequestState);
+        auto task = taskManager_->tasks();
+        auto prestoTask = task.find(scanTaskId)->second;
+        prestoTask->task = createDummyExecTask(scanTaskId, planFragment);
```

Comment:
This tests seems to be very tightly coupled with the TaskManager's implementation. I'm wondering if there is a way to test the desired behavior without such tight coupling.